spark/src/test/java/org/apache/iceberg/actions/TestRewriteDataFilesAction.java

```diff
@@ -311,6 +320,76 @@ public void testRewriteLargeTableHasResiduals() {
     Assert.assertEquals("Rows must match", records, actualRecords);
   }
 
+  @Test
+  public void testRewriteDataFilesForLargeFile() throws IOException, AnalysisException {
+    PartitionSpec spec = PartitionSpec.unpartitioned();
+    Map<String, String> options = Maps.newHashMap();
+    Table table = TABLES.create(SCHEMA, spec, options, tableLocation);
+    Assert.assertNull("Table must be empty", table.currentSnapshot());
+
+    List<Record> excepted = Lists.newArrayList();
+    Record baseRecord = GenericRecord.create(SCHEMA);
+
+    GenericAppenderFactory genericAppenderFactory = new GenericAppenderFactory(SCHEMA);
+    int count = 0;
+    File file = temp.newFile();
+    try (FileAppender<Record> fileAppender =
+            genericAppenderFactory.newAppender(Files.localOutput(file), FileFormat.PARQUET)) {
+      int fileSize = 10000;
+      for (; fileAppender.length() < fileSize; count++) {
+        Record record = baseRecord.copy();
+        record.setField("c1", count);
+        record.setField("c2", "foo" + count);
+        record.setField("c3", "bar" + count);
+        fileAppender.add(record);
+        excepted.add(record);
+      }
+    }
+
+    DataFile dataFile = DataFiles.builder(table.spec())
+        .withPath(file.getAbsolutePath())
+        .withFileSizeInBytes(file.length())
+        .withFormat(FileFormat.PARQUET)
+        .withRecordCount(count)
+        .build();
+
+    table.newAppend().appendFile(dataFile).commit();
+
+    List<ThreeColumnRecord> records1 = Lists.newArrayList(
+        new ThreeColumnRecord(1, "BBBBBBBBBB", "BBBB"),
+        new ThreeColumnRecord(1, "DDDDDDDDDD", "DDDD")
+    );
+    writeRecords(records1);
+
+    table.refresh();
+
+    CloseableIterable<FileScanTask> tasks = table.newScan().planFiles();
+    List<DataFile> dataFiles = Lists.newArrayList(CloseableIterable.transform(tasks, FileScanTask::file));
+    Assert.assertEquals("Should have 3 scan tasks before rewrite", 3, dataFiles.size());
+
+    spark.read().format("iceberg").load(tableLocation).createTempView("rows");
+    long originalNumRecords = spark.read().format("iceberg").load(tableLocation).count();
+    List<Object[]> originalRecords = sql("SELECT * from rows sort by c2");
+
+    Actions actions = Actions.forTable(table);
+
+    long targetSizeInBytes = file.length() - 10;
```

Comment:
It would be easier to understand if this were set up where `file` is used to create `dataFile`. Or, this could use the length of `dataFile` instead so we don't need to make sure `dataFile` and `file` have the same length.

Reply:
I just don’t understand this very well, I need to set splitTargetSize to a value smaller than the largest file